src/notes/components/Sidebar.tsx

```diff
@@ -24,6 +24,18 @@ const Sidebar = ({
   os, notes, canChangeOrder, onChangeOrder, active, width, onActivateNote, onNoteContextMenu, onNewNote, sync, openNoteOnMouseHover,
 }: SidebarProps): h.JSX.Element => {
   const sidebarRef = useRef<HTMLDivElement>(null);
+  const [onDraggingNoteOriginator, setOnDraggingNoteOriginator] = useState<string | undefined>(undefined);
+  const commonSidebarNotesProps: Omit<SidebarNotesProps, "id" | "notes" | "canDragEnter" | "onChangeOrder"> = {
+    active,
+    onActivateNote,
+    onNoteContextMenu,
+    canChangeOrder,
+    openNoteOnMouseHover,
+    setOnDraggingNoteOriginator,
```

Comment:
`onDraggingNoteOriginator` is used to make sure only notes within their group (pinned or unpinned) can be reordered.